src/libfetchers/git-lfs-fetch.hh

```diff
@@ -0,0 +1,587 @@
+        if (pattern_end == std::string::npos)
+            continue;
+
+        AttrRule rule;
+        rule.pattern = line.substr(0, pattern_end);
+        // Workaround for libgit2 matching issues when the pattern starts with a recursive glob
+        // These should effectively match the same files
+        // https://github.com/libgit2/libgit2/issues/6946
+        if (rule.pattern.starts_with("/**/")) {
+            rule.pattern = rule.pattern.substr(4);
+        }
+        while (rule.pattern.starts_with("**/")) {
+            rule.pattern = rule.pattern.substr(3);
+        }
+
+        git_strarray patterns = {0};
+        const char * pattern_str = rule.pattern.c_str();
+        patterns.strings = const_cast<char **>(&pattern_str);
+        patterns.count = 1;
+
+        if (git_pathspec_new(&rule.pathspec, &patterns) != 0) {
+            auto error = git_error_last();
+            std::stringstream ss;
+            ss << "git_pathspec_new parsing '" << line << "': " << (error ? error->message : "unknown error")
+               << std::endl;
+            warn(ss.str());
+            continue;
+        }
+
+        size_t attr_start = line.find_first_not_of(" \t", pattern_end);
+        if (attr_start != std::string::npos) {
+            std::string_view rest(line);
+            rest.remove_prefix(attr_start);
+
+            while (!rest.empty()) {
+                size_t attr_end = rest.find_first_of(" \t");
+                std::string_view attr = rest.substr(0, attr_end);
+
+                if (attr[0] == '-') {
+                    rule.attributes[std::string(attr.substr(1))] = "false";
+                } else if (auto equals_pos = attr.find('='); equals_pos != std::string_view::npos) {
+                    auto key = attr.substr(0, equals_pos);
+                    auto value = attr.substr(equals_pos + 1);
+                    rule.attributes[std::string(key)] = std::string(value);
+                } else {
+                    rule.attributes[std::string(attr)] = "true";
+                }
+
+                if (attr_end == std::string_view::npos)
+                    break;
+
+                rest = rest.substr(attr_end);
+                size_t next_attr = rest.find_first_not_of(" \t");
+                if (next_attr == std::string_view::npos)
+                    break;
+                rest = rest.substr(next_attr);
+            }
+        }
+
+        rules.push_back(std::move(rule));
+    }
+
+    return rules;
+}
+
+void Fetch::init(git_repository * repo, const std::string & gitattributesContent)
+{
+    const auto remoteUrl = lfs::getLfsEndpointUrl(repo);
+
+    this->gitUrl = parseGitUrl(remoteUrl);
+    if (this->gitUrl.protocol == "ssh") {
+        this->token = lfs::getLfsApiToken(this->gitUrl);
+    }
+    this->rules = lfs::parseGitAttrFile(gitattributesContent);
+    this->ready = true;
+}
+
+bool Fetch::hasAttribute(const std::string & path, const std::string & attrName, const std::string & attrValue) const
+{
```

Comment:
Do we have a compelling reason not to use [`git_attr_get_ext`](https://libgit2.org/docs/reference/main/attr/git_attr_get_ext.html)?

Reply:
Didn't implement this part, @b-camacho may have more context, but I could try my hand at using this instead to see if it still works.

Reply:
Not a satisfying explanation but I could not get it to work without materializing the files on disk, but I'll try again